### ISSUE TITLE
Blinking custom keyboard after switching between standard text fields

### ISSUE THREAD

**Issue by Jahn08** [diegoveloper/flutter_keyboard_actions#111]
Hello, Diego!  First and foremost, thank you a lot for your library - it's a great aid in creating any kind of keyboard in Flutter!

However, I ran into a problem when using a custom keyboard built with a mixin `KeyboardCustomPanelMixin<String>` on an Android device with **Flutter 1.17.4**. When you tap on a text field with a custom keyboard, then on another text field with a standard keyboard and again on the previous field the custom keyboard blinks and then disappears:

![blinking_custom_keyboard](https://user-images.githubusercontent.com/23010028/90865347-eedab200-e39a-11ea-8ec3-82d6360e5894.gif)

At first I thought the issue was related to one of the closed [Touching the input box on an Android device does not display the soft keyboard!](https://github.com/diegoveloper/flutter_keyboard_actions/issues/97). However, after downgrading my emulator image to **"system-images;android-29;google_apis;x86_64"** I still  experienced the same problem.

The issue seemed in a method `KeyboardActionstate.didChangeMetrics()`:
```
final value = WidgetsBinding.instance.window.viewInsets.bottom;
 if (value > 0) {
   _onKeyboardChanged(true);
   isKeyboardOpen = true;
 }
```
So, I created a descendant from your class and introduced a small "crutch" to make sure that the place is problematic:
```
 @override
 void didChangeMetrics() {
   final bottom = WidgetsBinding.instance.window.viewInsets.bottom;
   final hasFocus = FocusScope.of(context).hasFocus;

   if (bottom > 0 && hasFocus)
     super.didChangeMetrics();
 }
```
The behaviour changed, although the keyboard was now falling from the central part of the screen:
![falling_custom_keyboard](https://user-images.githubusercontent.com/23010028/90866421-8987c080-e39c-11ea-9cf2-2fe58143fe7f.gif)

Now, the custom keyboard appeared to be trying to render itself while the previous standard keyboard was still in view. So, the final code was:
```
class KeyboardActionsStateBottomed extends KeyboardActionstate {
    final FocusNode _focusNode;

    KeyboardActionsStateBottomed(FocusNode fieldFocusNode): _focusNode = fieldFocusNode, super() {
        _focusNode.addListener(() { 
            final bottom = WidgetsBinding.instance.window.viewInsets.bottom;
            final mainFocusNode = FocusScope.of(context);

            if (mainFocusNode.hasFocus && bottom > 0) {
                mainFocusNode.unfocus();

                new Timer(new Duration(milliseconds: 100), 
                    () => _focusNode.requestFocus());
            }
        });
    }
}
```
In other words, we wait for a short while to be certain that the old keyboard has already disappeared before drawing the custom one:

![amended_custom_keyboard](https://user-images.githubusercontent.com/23010028/90867303-df109d00-e39d-11ea-96cb-cabc3ac0f51b.gif)

Now it works as expected, although I have a feeling that there might be a better solution to the problem. Additionally, it probably has some side effects I haven't encountered yet.

**Comment by Lomski** [comment]
Have same issue. When I try to open custom keyboard it disappears if the standard keyboard already opened.
